src/core/specter-wallet.ts

```diff
@@ -201,8 +204,20 @@ export class SpecterWallet {
 
     const memo = `deposit of ${sats} sats to the cold storage wallet`
 
-    const outgoingOnchainTxns = await getOnChainTransactions({ lnd, incoming: false })
-    const [{ fee }] = outgoingOnchainTxns.filter((tx) => tx.id === id)
+    const getOnChainFee = async (txHash: OnChainTxHash): Promise<Satoshis> => {
+      const onChainService = OnChainService(TxDecoder(BTC_NETWORK))
+      if (onChainService instanceof Error) return toSats(0)
```

Comment:
not sure we should return 0 in case of an error?

Reply:
it actually depends if this is for a settleTransaction or for a request for fee from the user. we should probably not return 0 for the latter.

Reply:
I see this is in specter. nevermind.